py/htm/encoders/eye.py

```diff
@@ -30,58 +122,81 @@ class Eye:
 
     Attribute output_sdr ... retina's output
     Attribute roi ... The most recent view, kept as a attribute.
-    Attribute parvo ... 
-    Attribute magno ... 
+    Attribute parvo ... SDR with parvocellular pathway (color)
+    Attribute magno ... SDR with magnocellular pathway (movement) 
 
     The following three attributes control where the eye is looking within
     the image.  They are Read/Writable.
     Attribute position     (X, Y) coords of eye center within image
-    Attribute orientation  ... units are radians
-    Attribute scale        ... 
+    Attribute orientation  ... units are radians TODO of what? sensor/image
+    Attribute scale        ... TODO
     """
     def __init__(self,
-        output_diameter   = 200,
-        resolution_factor = 3,
-        fovea_scale       = .177,
-        sparsity          = .2,):
+        output_diameter   = 200, #TODO set as percentage of imput image?
```

Comment:
The output-diameter controls how many bits are in output encoded SDR, and I think that the output SDR should have a known size, irrespective of input image size.

Also, I intended for this encoder to be used a bit differently: the AI is supposed to control where in the image the eye is looking, so that the eye does not need to cover the whole image, it can search through the image.

Reply:
> and I think that the output SDR should have a known size, irrespective of input image size.

agree, I was actually thinking to use "standard" output SDR size, and use the diameter only internally. 

> the AI is supposed to control where in the image the eye is looking, so that the eye does not need to cover the whole image, 

sure, that's the point. Might have a member `pct_area_covered` by "one look" computed from diameter and img size. But you're right, biologically that does not make sense, as the scene is continuous, and we look (around) as needed.

Reply:
> pct_area_covered

The `scale` parameter should be used to control this value.

Reply:
Regarding the bits in output SDR, the ROI images (with blacked out corners, active inner circle), is the ROI only for visualization, or also the SDR is effectively not a square, but an inner circle?

Reply:
The Region Of Interest (ROI, pretty pictures) is showing what the eye is looking at. The ROI images have not been encoded.

The encoding process involves doing a log-polar transform on the ROI, which discards the blacked out areas and outputs a rectangular shaped image.  The log-polar transform is the reason why the outputs of the encoder (Parvo & magno cellular) are warped.

Technically, the blacked out areas are approximately where the log-polar transform cuts off, but I do not know an exact formula for the radius of the blacked out areas.